upgrade/change_v21.03.0.go

```diff
@@ -0,0 +1,449 @@
+ * limitations under the License.
+ */
+
+package upgrade
+
+import (
+	"fmt"
+	"math"
+	"net/http"
+	"strconv"
+
+	"github.com/dgraph-io/badger/v3"
+	"github.com/dgraph-io/dgo/v200"
+	"github.com/dgraph-io/dgo/v200/protos/api"
+	"github.com/dgraph-io/dgraph/graphql/schema"
+	"github.com/dgraph-io/dgraph/posting"
+	"github.com/dgraph-io/dgraph/protos/pb"
+	"github.com/dgraph-io/dgraph/x"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+)
+
+const (
+	queryCORS_v21_03_0 = `{
+			cors(func: has(dgraph.cors)){
+				uid
+				dgraph.cors
+			}
+		}
+	`
+	querySchema_v21_03_0 = `{
+			schema(func: has(dgraph.graphql.schema)){
+				uid
+				dgraph.graphql.schema
+			}
+		}
+	`
+	queryPersistedQuery_v21_03_0 = `{
+			pquery(func: type(dgraph.graphql.persisted_query)) {
+				uid
+				dgraph.graphql.p_query
+				dgraph.graphql.p_sha256hash
+			}
+		}
+	`
+)
+
+type cors struct {
+	UID  string   `json:"uid"`
+	Cors []string `json:"dgraph.cors,omitempty"`
+}
+
+type sch struct {
+	UID       string `json:"uid"`
+	GQLSchema string `json:"dgraph.graphql.schema"`
+}
+
+type pquery struct {
+	UID   string `json:"uid"`
+	Query string `json:"dgraph.graphql.p_query,omitempty"`
+	SHA   string `json:"dgraph.graphql.p_sha256hash,omitempty"`
+}
+
+func updateGQLSchema(jwt *api.Jwt, gqlSchema string, corsList []string) error {
+	if len(gqlSchema) == 0 || len(corsList) == 0 {
+		fmt.Println("Nothing to update in GraphQL shchema. Either schema or cors not found.")
+		return nil
+	}
+	gqlSchema += "\n\n\n# Below schema elements will only work for dgraph" +
+		" versions >= 21.03. In older versions it will be ignored."
+	for _, c := range corsList {
+		gqlSchema += fmt.Sprintf("\n# Dgraph.Allow-Origin \"%s\"", c)
+	}
+
+	// Update the schema.
+	header := http.Header{}
+	header.Set("X-Dgraph-AccessToken", jwt.AccessJwt)
+	header.Set("X-Dgraph-AuthToken", Upgrade.Conf.GetString(authToken))
+	updateSchemaParams := &GraphQLParams{
+		Query: `mutation updateGQLSchema($sch: String!) {
+			updateGQLSchema(input: { set: { schema: $sch }}) {
+				gqlSchema {
+					id
+				}
+			}
+		}`,
+		Variables: map[string]interface{}{"sch": gqlSchema},
+		Headers:   header,
+	}
+
+	adminUrl := Upgrade.Conf.GetString(alphaHttp) + "/admin"
+	resp, err := makeGqlRequest(updateSchemaParams, adminUrl)
+	if err != nil {
+		return err
+	}
+	if len(resp.Errors) > 0 {
+		return errors.Errorf("Error while updating the schema %s\n", resp.Errors.Error())
+	}
+	fmt.Println("Successfully updated the GraphQL schema.")
+	return nil
+}
+
+var deprecatedPreds = map[string]struct{}{
+	"dgraph.cors":                      {},
+	"dgraph.graphql.schema_created_at": {},
+	"dgraph.graphql.schema_history":    {},
+	"dgraph.graphql.p_sha256hash":      {},
+}
+
+var deprecatedTypes = map[string]struct{}{
+	"dgraph.type.cors":       {},
+	"dgraph.graphql.history": {},
+}
+
+func dropDeprecated(dg *dgo.Dgraph) error {
+	if !Upgrade.Conf.GetBool(deleteOld) {
```

Comment:
why deleteOld. Does this have any relation to upgrade from 20.11 to 21.03?

Reply:
This `deleteOld` is used when upgrading from other versions as well.